src/txorphanage.cpp

```diff
@@ -15,14 +14,35 @@ static constexpr int64_t ORPHAN_TX_EXPIRE_TIME = 20 * 60;
 /** Minimum time between orphan transactions expire time checks in seconds */
 static constexpr int64_t ORPHAN_TX_EXPIRE_INTERVAL = 5 * 60;
 
+void TxOrphanage::SubtractOrphanBytes(unsigned int size, NodeId peer)
+{
+    AssertLockHeld(m_mutex);
+    Assume(m_peer_bytes_used.count(peer) > 0);
+    Assume(m_peer_bytes_used.at(peer) >= size);
+    m_peer_bytes_used.at(peer) -= size;
+    if (m_peer_bytes_used.at(peer) == 0) {
+        m_peer_bytes_used.erase(peer);
+    }
+}
 
 bool TxOrphanage::AddTx(const CTransactionRef& tx, NodeId peer)
 {
     LOCK(m_mutex);
+    if (tx == nullptr) {
+        Assume(false);
+        return false;
+    }
```

Comment:
Would be cleaner to write:
```suggestion
    if (!Assume(tx)) {
        return false;
    }
```

Reply:
Done